### PR TITLE
Unit test for scheduling a Mma op where input B has an allocation domain.

### PR DESCRIPTION
**Pull request by protonu**
As a first step to supporting allocation domain in our matmul scheduler, this unit test demonstrates how we can use reordering to extend scheduling a Mma op when the input Tensorview (B) has an allocation domain.

The next PR on top this will extend the scheduler to support a strided input B in a unit test that schedules a matmul.

### CODE REVIEW

**Comment by jacobhinkle** [comment]
Strided here means transposed but still contiguous right? Specifically you are setting the stride order on the input and verifying that we compute the matmul properly.

**Comment by protonu** [comment]
> Strided here means transposed but still contiguous right? Specifically you are setting the stride order on the input and verifying that we compute the matmul properly.

Yes, that's correct. I updated the title to just mention an allocation domain.

File: tests/cpp/test_mma.cpp [hunk 1]
```diff
@@ -85,43 +88,79 @@ class MmaTest : public NVFuserFixtureParamTest<MmaTestParams> {
...
+    std::unordered_map<int64_t, int64_t> reorder_map;
+    int64_t idx = 0;
+    std::transform(
+        b_permutation.begin(),
+        b_permutation.end(),
+        std::inserter(reorder_map, reorder_map.end()),
+        [&idx](const int64_t v) { return std::make_pair(idx++, v); });
+
+    // Reorder the ouput of Mma.
+    tv2->reorder(reorder_map);
```
Comment (zasdfgbnm) [inline tests/cpp/test_mma.cpp:145]: Should we just add a `TensorView::reorder(b_permutation)`?

**Review by zasdfgbnm — APPROVED** [review]

**Comment by protonu** [comment]
!build

**Review by zasdfgbnm — APPROVED** [review]

**Pull request merged**